starboard/android/apk/app/src/main/java/dev/cobalt/media/AudioTrackBridge.java

```diff
@@ -165,15 +166,38 @@ public AudioTrackBridge(
       audioTrackBufferSize /= 2;
     }
     if (isAudioTrackValid()) {
-      audioTrackBufferSize = audioTrack.getBufferSizeInFrames() * channelCount * getBytesPerSample(sampleType);
+      // If the AudioTrack encoding indicates compressed data,
+      // e.g. AudioFormat.ENCODING_AC3, then the frame count returned
+      // is the size of the AudioTrack buffer in bytes.
+      audioTrackBufferSize = audioTrack.getBufferSizeInFrames();
+      switch (sampleType) {
+        case AudioFormat.ENCODING_PCM_16BIT:
+          sampleTypeString = "ENCODING_PCM_16BIT";
+          audioTrackBufferSize *= getBytesPerSample(sampleType) * channelCount;
+          break;
+        case AudioFormat.ENCODING_PCM_FLOAT:
+          sampleTypeString = "ENCODING_PCM_FLOAT";
+          audioTrackBufferSize *= getBytesPerSample(sampleType) * channelCount;
+          break;
+        case AudioFormat.ENCODING_AC3:
+          sampleTypeString = "ENCODING_AC3";
+          break;
+        case AudioFormat.ENCODING_E_AC3:
+          sampleTypeString = "ENCODING_E_AC3";
+          break;
+        default:
+          Log.i(TAG, String.format("Unknown AudioFormat %d.", sampleType));
+          break;
+      }
     }
     Log.i(
         TAG,
         "AudioTrack created with buffer size %d (preferred: %d).  The minimum buffer size is"
```

Comment:
nit: "AudioTrack created with format (%s) and buffer size %d (preferred: %d), ..."  It's usually great to write the log as readable sentences or phrases.

Reply:
Fixed.